src/App.tsx

```diff
@@ -15,47 +15,94 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  if (sortType) {
+    visibleGoods.sort((goodA, goodB) => (sortType === SortType.ALPHABET
+      ? goodA.localeCompare(goodB)
+      : goodB.length - goodA.length));
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+const defaultSort: ReorderOptions = {
+  sortType: SortType.NONE,
+  isReversed: false,
+};
+
 export const App: React.FC = () => {
+  const [sortState, setSortState] = useState<ReorderOptions>(defaultSort);
+
+  const onSortClick = (newSortType: SortType) => setSortState(
+    prev => ({ ...prev, sortType: newSortType }),
+  );
+
+  const onReversClick = () => setSortState((prev) => (
```

Comment:
Consider renaming `onReversClick` to `handleReverseClick` for better consistency with React event handlers naming conventions and fix the typo in the name.